src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsApproverPage.tsx

```diff
@@ -1,45 +1,242 @@
 import * as Policy from '@userActions/Policy/Policy';
+import * as Workflow from '@userActions/Workflow';
 import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
 import type SCREENS from '@src/SCREENS';
-import type {PersonalDetailsList, PolicyEmployee} from '@src/types/onyx';
+import type {Beta, PolicyEmployee} from '@src/types/onyx';
+import type {Icon} from '@src/types/onyx/OnyxCommon';
 import {isEmptyObject} from '@src/types/utils/EmptyObject';
 
 type WorkspaceWorkflowsApprovalsApproverPageOnyxProps = {
-    /** All of the personal details for everyone */
-    personalDetails: OnyxEntry<PersonalDetailsList>;
+    /** Beta features list */
+    // eslint-disable-next-line react/no-unused-prop-types -- This prop is used in the component
+    betas: OnyxEntry<Beta[]>;
 };
 
 type WorkspaceWorkflowsApprovalsApproverPageProps = WorkspaceWorkflowsApprovalsApproverPageOnyxProps &
     WithPolicyAndFullscreenLoadingProps &
     StackScreenProps<FullScreenNavigatorParamList, typeof SCREENS.WORKSPACE.WORKFLOWS_APPROVALS_APPROVER>;
+
+type SelectionListApprover = {
+    text: string;
+    alternateText: string;
+    keyForList: string;
+    isSelected: boolean;
+    login: string;
+    rightElement?: React.ReactNode;
+    icons: Icon[];
+};
+type ApproverSection = SectionListData<SelectionListApprover, Section<SelectionListApprover>>;
+
+function WorkspaceWorkflowsApprovalsApproverPage(props: WorkspaceWorkflowsApprovalsApproverPageProps) {
+    if (Permissions.canUseWorkflowsAdvancedApproval(props.betas) && props.route.params.approverIndex !== undefined) {
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        return <WorkspaceWorkflowsApprovalsApproverPageNew {...props} />;
+    }
+
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    return <WorkspaceWorkflowsApprovalsApproverPageOld {...props} />;
+}
+
+function WorkspaceWorkflowsApprovalsApproverPageNew({policy, personalDetails, isLoadingReportData = true, route}: WorkspaceWorkflowsApprovalsApproverPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const [didScreenTransitionEnd, setDidScreenTransitionEnd] = useState(false);
+    const [searchTerm, debouncedSearchTerm, setSearchTerm] = useDebouncedState('');
+    const [approvalWorkflow, approvalWorkflowMetadata] = useOnyx(ONYXKEYS.APPROVAL_WORKFLOW);
+    const [selectedApproverEmail, setSelectedApproverEmail] = useState<string | undefined>(undefined);
+
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    const shouldShowNotFoundView = (isEmptyObject(policy) && !isLoadingReportData) || !PolicyUtils.isPolicyAdmin(policy) || PolicyUtils.isPendingDeletePolicy(policy);
+    const approverIndex = route.params.approverIndex ?? 0;
+
+    useEffect(() => {
+        const currentApprover = approvalWorkflow?.approvers[approverIndex];
```

Comment:
For creation flow we could use `approverEmail` instead of `approverIndex`, however when editing an existing approval workflow (which will use the same approver page) we need an information which approver is currently edited, sole `approverEmail` would not be enough as there might be a rare ocasion where there are duplicated emails in one workflow.

Fixed in https://github.com/Expensify/App/pull/46510/commits/6d6a4a67df7fa71bdc51df16a3d4e2f8e11ee81c. My logic is simple:
- When selecting an approver user have to click on 'Next' button to set the approver in the onyx
  - Creation flow: When going back to "expenses from" screen, the selected approver will be cleared
  - Edition flow: When going back to "edit approval workflow" screen, the selected approver won't be cleared

Reply:
Let me know what do you think @suneox 
https://github.com/user-attachments/assets/af1e3953-3e9b-4318-ba5b-2fe09639d523